x-pack/filebeat/input/salesforce/input_manager_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package salesforce
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/beats/v7/libbeat/statestore/storetest"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/go-concert/unison"
+)
+
+// compile-time check if querier implements InputManager
+var _ v2.InputManager = InputManager{}
+
+func makeTestStore(data map[string]interface{}) *statestore.Store {
+	memstore := &storetest.MapStore{Table: data}
+	reg := statestore.NewRegistry(&storetest.MemoryStore{
+		Stores: map[string]*storetest.MapStore{
+			"test": memstore,
+		},
+	})
+	store, err := reg.Get("test")
+	if err != nil {
+		panic("failed to create test store")
+	}
+	return store
+}
+
+type stateStore struct{}
+
+func (stateStore) Access() (*statestore.Store, error) {
+	return makeTestStore(map[string]interface{}{"hello": "world"}), nil
+}
+func (stateStore) CleanupInterval() time.Duration { return time.Duration(0) }
+
+// compile-time check if stateStore implements cursor.StateStore
+var _ cursor.StateStore = stateStore{}
+
+func TestInputManager(t *testing.T) {
+	inputManager := NewInputManager(logp.NewLogger("salesforce_test"), stateStore{})
+
+	var inputTaskGroup unison.TaskGroup
+	defer func() {
```

Comment:
Using defer like this is helpful for multi-line statements in a function that needs to be deferred. 
If the return value is not being used, it's better to use it like this:
```suggestion
	defer inputTaskGroup.Stop() 
```